src/main/java/org/jenkinsci/modules/windows_slave_installer/WindowsSlaveInstaller.java

```diff
@@ -77,17 +91,23 @@ static int runElevated(File agentExe, String command, TaskListener out, File pwd
         }
     }
 
-
-    @Override @SuppressFBWarnings("DM_EXIT")
+    @Override
     public void install(LaunchConfiguration params, Prompter prompter) throws InstallationException, IOException, InterruptedException {
-        if(!DotNet.isInstalled(2,0))
+        install(params, prompter, false);
+    }
+    
+    @SuppressFBWarnings(value = "DM_EXIT", justification = "Legacy design, but as designed")
+    /*package*/ void install(LaunchConfiguration params, Prompter prompter, boolean mock) throws InstallationException, IOException, InterruptedException {
+        if(!mock && !DotNet.isInstalled(2,0)) {
```

Comment:
No, it does not. I consider proposing the .NET 2 deprecation in Jenkins, but it's rather for WinSW 3